src/main/resources/Development-documentation/gerrit-scenarios.md

```diff
@@ -0,0 +1,257 @@
+  - [Create patch set with no previous events - CASE4](#create-patch-set-with-no-previous-events---case4)
+
+<!-- /TOC -->
+
+Glossary:
+
+- **SCS** SourceChangeSubmitted
+- **SCC** SourceChangeCreated
+- **E?** Events
+- **C?** Commits
+- **P?** Push commands
+
+## Create patch set - CASE1
+
+Scenario overview:
+
+This scenario describes a standard review cycle where a user uploads a patch
+set, receives some comments, corrects the comments and submits the changes. The
+plugin can find the SCS event from the previous successful review.
+
+Preconditions:
+
+- Previous review submitted to the master with hash `C0`
+- The plugin sends SCS with id `E0`
+
+The user does the following:
+
+- Creates a branch
+- Updates the code
+- Squashes the commits to one commits (`C1`)
+- Pushes to `refs/for/[branch name]` (`P1`)
+- Receives comments from reviewer
+- Updates the code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]` (`P2`)
+- Gets ok from the reviewer
+- Hits the submit button in Gerrit
+
+Eiffel events sent from the plugin:
+
+- SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
+- SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
+- SCS(`E3`) at submit with `CHANGE` link set `E2`
+
+Commit history after submit:
+
+```git
+* C2 Review changes
+* C0 Commit from previous user
+```
+
+Trigger table:
+
+| Event | Triggered by | Comment |
+| ---   | ---          | ---     |
+| E0    |              | Submit  |
+| E1    | P1           |
+| E2    | P2           |
+| E3    |              | Submit  |
+
+SCC Event table:
+
+| Event | BASE | PREVIOUS_VERSION |
+| ---   | ---  | ---              |
+| E1    | E0   |
+| E2    | E0   | E1               |
+
+## Create patch set with rebase - CASE2
+
+Scenario overview:
+
+This scenario describes a workflow where a user performs a standard review
+cycle. Before the review finishes another review finishes and submits to
+master(`C01`).
+
+Preconditions:
+
+- Previous review submitted to the master with hash `C0`
+- The plugin sends SCS with id `E0`
+
+The user does the following:
+
+- Creates a branch
+- Updates the code
+- Squashes the commits to one commits (`C1`)
+- Pushes to `refs/for/[branch name]` (`P1`)
+- Receives comments from reviewer
+- Updates the code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]` (`P2`)
+- Gets ok from the reviewer
+- Forced to rebase the code due to `C01` (`C3`)  event `E01`
+- Pushes to `refs/for/[branch name]`  (`P3`)
+- Hits the submit button in Gerrit
+
+Eiffel events sent from the plugin:
+
+- SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
+- SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
+- SCC(`E3`) sent for `P3` push with `PREVIOUS_VERSION` link set `E2` and `BASE` link set to `E01`
+- SCS(`E4`) sent at submit with `CHANGE` link set `E3`
```

Comment:
What about PREVIOUS_VERSION for the SCS event? Will that be set?

Reply:
Fixed in f84b35e